python/hsfs/engine/python.py

```diff
@@ -1440,40 +1448,40 @@ def _convert_feature_log_to_df(feature_log, cols):
 
             return pd.DataFrame(feature_log, columns=cols)
         else:
-            return feature_log.copy(deep=False)
+            if isinstance(feature_log, pl.DataFrame):
+                return feature_log.clone().to_pandas()
+            elif isinstance(feature_log, pd.DataFrame):
+                return feature_log.copy(deep=False)
 
     @staticmethod
     def get_feature_logging_df(
-        fg,
-        features,
-        fg_features: List[TrainingDatasetFeature],
-        td_predictions: List[TrainingDatasetFeature],
-        td_col_name,
-        time_col_name,
-        model_col_name,
-        prediction=None,
-        training_dataset_version=None,
+        features: Union[pd.DataFrame, list[list], np.ndarray],
+        fg: FeatureGroup = None,
+        td_features: List[str] = None,
+        td_predictions: List[TrainingDatasetFeature] = None,
+        td_col_name: Optional[str] = None,
+        time_col_name: Optional[str] = None,
+        model_col_name: Optional[str] = None,
+        predictions: Optional[Union[pd.DataFrame, list[list], np.ndarray]] = None,
+        training_dataset_version: Optional[int] = None,
         hsml_model=None,
     ) -> pd.DataFrame:
-        import uuid
-
         features = Engine._convert_feature_log_to_df(
-            features, [f.name for f in fg_features]
+            features, td_features
         )
         if td_predictions:
-            prediction = Engine._convert_feature_log_to_df(
-                prediction, [f.name for f in td_predictions]
+            predictions = Engine._convert_feature_log_to_df(
```

Comment:
Do we need `Engine.` here? I see it removed in the next line for `cast_column_to_offline_type`

Reply:
yes because `get_feature_logging_df` is static method